### PR TITLE
Adds Tests for Untested RemoveSnapshot Pathway

### PR DESCRIPTION
**Pull request by RussellSpitzer**
This is an easy PR, no Prod code changes, just improves test coverage :)

Previously none of the tests in TestRemoveSnapshots used the code path for
valid snapshots which contained references to manifests which were created
in a snapshot that had previously expired. Although there are no bugs in that
logic, we add two tests which check this pathway to protect against future
breakage.

### CODE REVIEW

**Comment by RussellSpitzer** [comment]
cc @aokolnychyi

File: core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java [hunk 1]
```diff
@@ -356,6 +356,74 @@ public void testRetainZeroSnapshots() {
         () -> table.expireSnapshots().retainLast(0).commit());
   }
 
+  @Test
+  public void testScanExpireddManifestInValidSnapshotAppend() {
```
Comment (aokolnychyi) [inline core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java:360]: nit: `Expiredd`

Comment (RussellSpitzer) [reply core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java]: I'm gonna try size 18 font now .... :P

**Comment by aokolnychyi** [comment]
Thanks, @RussellSpitzer!

**Pull request merged**